packages/framework/src/Model/Product/Parameter/ParameterRepository.php

```diff
@@ -548,4 +548,23 @@ public function getProductsByParameterValues(array $parameterValues): array
             ->getQuery()
             ->execute();
     }
+
+    /**
+     * @param string $name
+     * @param string $locale
+     * @return \Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter|null
+     */
+    public function findParameterByName(string $name, string $locale): ?Parameter
+    {
+        return $this->em->createQueryBuilder()
+            ->select('p')
+            ->from(Parameter::class, 'p')
+            ->join('p.translations', 'pt')
+            ->where('pt.name = :name')
+            ->andWhere('pt.locale = :locale')
+            ->setParameter('name', $name)
+            ->setParameter('locale', $locale)
+            ->getQuery()
+            ->getOneOrNullResult();
```

Comment:
if existing data already contains the duplicities, this validator will result in the error 500, or?

Reply:
good catch, fixed in f9a776139d